src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ExplicitConversionSymbolReferenceFinder.UnderlyingNamedTypeVisitor.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.FindSymbols.Finders
+{
+    internal partial class ExplicitConversionSymbolReferenceFinder
+    {
+        private class UnderlyingNamedTypeVisitor : SymbolVisitor<INamedTypeSymbol?>
+        {
+            public static readonly UnderlyingNamedTypeVisitor Instance = new UnderlyingNamedTypeVisitor();
+
+            private UnderlyingNamedTypeVisitor()
+            {
+            }
+
+            public override INamedTypeSymbol? VisitArrayType(IArrayTypeSymbol symbol)
+                => Visit(symbol.ElementType);
+
+            public override INamedTypeSymbol? VisitDynamicType(IDynamicTypeSymbol symbol)
+                => null;
+
+            public override INamedTypeSymbol? VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
+                => null;
+
+            public override INamedTypeSymbol? VisitPointerType(IPointerTypeSymbol symbol)
+                => Visit(symbol.PointedAtType);
+
+            public override INamedTypeSymbol? VisitTypeParameter(ITypeParameterSymbol symbol)
+                => null;
+
+            public override INamedTypeSymbol? VisitNamedType(INamedTypeSymbol symbol)
+                => symbol;
+
+            public override INamedTypeSymbol? DefaultVisit(ISymbol symbol)
+            {
+                Debug.Fail($"Symbol case not handled: {symbol.Kind}");
```

Comment:
(I thought I sent this last night, but found I had the tab open without hitting approve... >.>)

Reply:
In a case like this, it would likely be because teh compiler added a new 'type symbol type' (for example, we did that when we added func-ptrs).  In a case liek this, we'd then decide how teh feature should work and update this code accordingly.